compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -266,12 +266,11 @@ trait ParallelTesting extends RunnerOrchestration { self =>
      */
     final def diffTest(testSource: TestSource, checkFile: JFile, actual: List[String], reporters: Seq[TestReporter], logger: LoggedRunnable) = {
       for (msg <- FileDiff.check(testSource.title, actual, checkFile.getPath)) {
-        onFailure(testSource, reporters, logger, Some(msg))
-
         if (updateCheckFiles) {
           FileDiff.dump(checkFile.toPath.toString, actual)
           echo("Updated checkfile: " + checkFile.getPath)
         } else {
+          onFailure(testSource, reporters, logger, Some(msg))
```

Comment:
Originally I left this one out to force users to re-run the tests against the new checkfile. This can be useful if there is a source of non-determinism in the output.

Reply:
I can revert this. Was pretty confusing initially but I might be biased by the Scala 2 experience.

Reply:
You can keep your change. There is no right or wrong way to do this. Your way might be easier to understand.